clients/pkg/promtail/client: NPE if metrics not provided

If you invoke client.New with a nil *Metrics object, you get a nasty panic

```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x48 pc=0x1059a1a68]

goroutine 70 [running]:
testing.tRunner.func1.2({0x105eb57a0, 0x106873d70})
	/Users/kburke/go/src/testing/testing.go:1545 +0x1c8
testing.tRunner.func1()
	/Users/kburke/go/src/testing/testing.go:1548 +0x360
panic({0x105eb57a0?, 0x106873d70?})
	/Users/kburke/go/src/runtime/panic.go:765 +0x124
github.com/grafana/loki/clients/pkg/promtail/client.newClient(_, {{0x0, 0x0}, {0x140006d8bd0}, 0x0, 0x0, {0x0, 0x0, 0x0, {0x0, ...}, ...}, ...}, ...)
	/path/to/clients/pkg/promtail/client/client.go:235 +0x338
```

If a Metrics object is a requirement, instead of a panic it would be good to check for a nil *Metrics - the same way we check for a target URL - and return an appropriate error.

--- a/clients/pkg/promtail/client/client.go
+++ b/clients/pkg/promtail/client/client.go
@@ -197,6 +197,9 @@ func newClient(metrics *Metrics, cfg Config, maxStreams, maxLineSize int, maxLin
 	if cfg.URL.URL == nil {
 		return nil, errors.New("client needs target URL")
 	}
+	if metrics == nil {
+		return nil, errors.New("metrics must be instantiated")
+	}
 
 	ctx, cancel := context.WithCancel(context.Background())
 

